File: Programmers/연습문제/Level 1/2016년.java
/* Programmers
 * 연습문제 - 2016년
 * created on 2021.01.18
 * created by jionchu */

class Solution {
    public String solution(int a, int b) {
        int days = b;
        
        //일수 계산
        for (int i=0;i<a;i++) {
            switch(i) {
                case 0: break;
                case 1: case 3: case 5: case 7: case 8: case 10: case 12: days += 31; break;
                case 2: days += 29; break;
                case 4: case 6: case 9: case 11: days += 30; break;
            }
        }
        
        //요일 계산
        switch (days%7) {
            case 0: return "THU";
            case 1: return "FRI";
            case 2: return "SAT";
            case 3: return "SUN";
            case 4: return "MON";
            case 5: return "TUE";
            case 6: return "WED";
            default: return "FRI";
        }
    }
}

File: Programmers/연습문제/Level 1/2016년.kt
/* Programmers
 * 연습문제 - 2016년
 * created on 2021.06.16
 * created by jionchu */

class Solution {
    fun solution(a: Int, b: Int): String {
        
        //요일 배열
        val dayOfWeek = arrayOf("SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT")
        //월별 일수 배열
        val daysOfMonth = arrayOf(31,29,31,30,31,30,31,31,30,31,30,31)
        
        //1월 1일부터 a월 b일까지의 일수
        //1월 1일이 금요일(dayOfWeek[4])이므로 +4
        var days = 4 + b
        
        //월별 일수 더하기
        for (i in 2..a) {
            days += daysOfMonth[i-2]
        }
        
        //해당 날의 요일 return
        return dayOfWeek[days%7]
    }
}